src/NServiceBus.Core/Persistence/Learning/SagaPersister/LearningStorageAdapter.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace NServiceBus
+{
+    using System.Threading.Tasks;
+    using Extensibility;
+    using Outbox;
+    using Persistence;
+    using Transport;
+
+    class LearningStorageAdapter : ISynchronizedStorageAdapter
```

Comment:
Is `ISynchronizedStorageAdapter` specific to sagas? Based on the current name of the type, `LearningStorageAdapter` I'm thinking it isn't. Would it make sense to move this up into the base Learning folder? 

If not, I think the name should have Saga in it somewhere.